pkg/visibility/api/rest/pending_workloads_cq.go

```diff
@@ -66,6 +67,10 @@ func (m *pendingWorkloadsInCqREST) Get(ctx context.Context, name string, opts ru
 
 	wls := make([]v1alpha1.PendingWorkload, 0, limit)
 	pendingWorkloadsInfo := m.queueMgr.PendingWorkloadsInfo(name)
+	if pendingWorkloadsInfo == nil {
```

Comment:
Is the nil equivalent to non-existing ClusterQueue, or could this also happen when there are no pending workloads? I think the NotFound should only be returned when no ClusterQueue. I think it makes sense to have a test case for an existing cluster queue, but without any workloads.